relayer/src/chain/cosmos.rs

```diff
@@ -276,9 +276,25 @@ impl CosmosSdkChain {
 
         let response = self.block_on(broadcast_tx_sync(self, tx_bytes))?;
 
-        debug!("[{}] send_tx: broadcast_tx_sync: {:?}", self.id(), response);
-
-        self.incr_account_sequence()?;
+        match response.code {
+            tendermint::abci::Code::Ok => {
+                // A success means the account s.n. was increased
```

Comment:
Just a question - does a failure during DeliverTx also result in incrementing the account sequence? And is it possible that CheckTx succeeds while DeliverTx fails? 
I'm asking because I notice from my tests that a failure in DeliverTx (with a non-zero code) is still interpreted as `tendermint::abci::Code::Ok`.

Reply:
>does a failure during DeliverTx also result in incrementing the account sequence?

A failure in CheckTx means the transactions was not accepted into the mempool. The s.n., remains unchanged. A success in CheckTx (and I think regardless of the outcome of DeliverTx) means the s.n. increased.

> And is it possible that CheckTx succeeds while DeliverTx fails?

Yes, for instance DeliverTx can result in an error with insufficient fees, or "already executed" error.